### PR TITLE
kconfig: fix (re)definitions in Kconfig.defconfig files

### PR DESCRIPTION
**Pull request by gmarull**
Kconfig symbols must not be defined in Kconfig.defconfig.* files. Such
files are meant to change the default value of certain options.

This PR resolves such issues and introduces a new compliance check to prevent similar cases in the future.

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]
Thanks for cleaning this up.

Maybe a simple search in all `Kconfig.defconfig` for text strings in quotes `"Some text"` should be possible, cause no defconfigs ought to have prompts.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by gmarull** [comment]
@mbolivar-nordic @tejlmand please re-review.

**Review by tejlmand — APPROVED** [review]
thanks for this cleanup.

**Review by stephanosio — APPROVED** [review]
Nice work

**Review by MaureenHelm — APPROVED** [review]

**Review by ChiHuaL — APPROVED** [review]

File: boards/arm/bl654_usb/Kconfig.defconfig [hunk 1]
```diff
@@ -26,24 +26,14 @@ config FLASH_LOAD_OFFSET
 config USB_CDC_ACM
 	default n if USB_DEVICE_BLUETOOTH
 
-config BL654_USB_SERIAL_BACKEND_CDCACM
```
Comment (jfischer-no) [inline boards/arm/bl654_usb/Kconfig.defconfig:29]: It is not just shortcut, please move this option to boards/arm/bl654_usb/Kconfig, or drop this patch and assign an issue to me, thanks.

Comment (gmarull) [reply boards/arm/bl654_usb/Kconfig.defconfig:29]: thanks, moved to Kconfig.

**Review by jfischer-no — APPROVED** [review]
:+1:

**Review by mbolivar-nordic — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**